rpc/openapi/openapi.yaml

```diff
@@ -2742,6 +2740,33 @@ components:
           type: string
           example: "Dialing seeds in progress. See /net_info for details"
 
+    BlockSearchResponse:
```

Comment:
I don't think we should add this without the corresponding implementation.

Reply:
It's already on `main` (i.e. in v0.34): https://github.com/tendermint/tendermint/blob/4309f54349fd90b03fe38271e8b19cfed2cfb2b6/rpc/core/types/responses.go#L209

While fixing the linting issues in the OpenAPI config, I discovered that there were a bunch of things that were missing/incorrect. So I've fixed them here to match the code on `main`.

See https://github.com/tendermint/tendermint/blob/4309f54349fd90b03fe38271e8b19cfed2cfb2b6/rpc/openapi/openapi.yaml#L1067 on `main` for example, where there's no definition of `BlockSearchResponse` in the OpenAPI config.